src/data/matrix/hadamard.lean

```diff
@@ -0,0 +1,141 @@
+/-
+Copyright (c) 2021 Lu-Ming Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Lu-Ming Zhang
+-/
+import data.matrix.notation
+import linear_algebra.matrix.trace
+import data.complex.basic
+
+/-!
+# Hadamard product of matrices
+
+This file defines the Hadamard product `matrix.Hadamard`
+and contains basic properties about them.
+
+## Main definition
+
+- `matrix.Hadamard`: defines the Hadamard product,
+  which is the pointwise product of two matrices of the same size.
+
+## Notation
+
+* `⊙`: the Hadamard product `matrix.Hadamard`;
+
+## References
+
+*  <https://en.wikipedia.org/wiki/Hadamard_product_(matrices)>
+
+## Tags
+
+Hadamard product, Hadamard
+-/
+
+variables {α β γ I J K L M N: Type*}
+variables {R : Type*}
+variables {m n p q r s t: ℕ}
+
+namespace matrix
+open_locale matrix big_operators
+open complex
+
+/-- `matrix.Hadamard` defines the Hadamard product,
+    which is the pointwise product of two matrices of the same size.-/
+@[simp]
+def Hadamard [has_mul α] (A : matrix I J α) (B : matrix I J α) : matrix I J α
+| i j := A i j * B i j
+
+localized "infix ` ⊙ `:100 := matrix.Hadamard" in matrix
+
+section basic_properties
+
+variables (A : matrix I J α) (B : matrix I J α) (C : matrix I J α)
+
+/- commutativity -/
+lemma Hadamard_comm [comm_semigroup α] : A ⊙ B = B ⊙ A :=
+ext $ λ _ _, mul_comm _ _
```

Comment:
This approach will work for every lemma from here to `zero_Hadamard` I think

Reply:
changed up to `smul_hadamard`. Not sure if it can be applied to others